pkgs/dart_mcp_server/bin/main.dart

```diff
@@ -40,6 +40,7 @@ void main(List<String> args) async {
             ),
         forceRootsFallback: parsedArgs.flag(forceRootsFallback),
         sdk: Sdk.find(dartSdkPath: dartSdkPath, flutterSdkPath: flutterSdkPath),
+        analytics: null,
```

Comment:
Can we actually create the Analytics instance here?

Reply:
If we are allowed to that would simplify things, otherwise we need a custom entry point in the SDK which does create one.

I just don't know if it is kosher or not, it feels a bit weird to be putting the current tool as the `dartTool`, when this is actually a binary that can be ran in other ways (from source, or as a git dep).

Reply:
Another option would be a command line flag to enable analytics.

Reply:
I don't think so, we will just get the dart native runtime as the resolved executable afaik.

Reply:
Hmm. Then if we need to use the Dart CLI as the parent tool, adding a hidden flag that is set from the dart mcp-server command only could be a good workaround. We should get clarity on that first from Ander though.